src/main/java/jenkins/branch/OrganizationFolder.java

```diff
@@ -1436,19 +1436,14 @@ private void completeNew(MultiBranchProjectFactory factory, Map<String, Object>
                                 .println("Ignoring duplicate child " + projectName + " named " + folderName);
                         return;
                     }
-                    MultiBranchProject<?, ?> project;
-                    try (ChildNameGenerator.Trace trace = ChildNameGenerator.beforeCreateItem(
```

Comment:
`AbstractWorkflowMultiBranchProjectFactory#doCreateProject` does not attempt to save the project, and both `SCMSourceObserver.ProjectObserver#completeNew` and `SCMSourceObserver.ProjectObserver#completeExisting` use `BulkChange`, so `idealNames` is never consulted here.